### DISCUSSION TITLE
Recommendations

### DISCUSSION THREAD

**Discussion by nikhilCad** [z-huang/music#456] [Ideas]
https://github.com/vfsfitvnm/ViMusic/issues/413

This implementation of recommendations described above can be implemented here as well. The "Explore" section just remains the same for me and just shows whatever trending songs in my country are. 

To mimic "personal recommendations", the above approach can be useful in which you first calculate the most popular song listen by user, and then give : Similar songs, artists, albums and playlists for it. 

Maybe a mix and match is also possible? Fetch all similar songs/artists/albums/playlists from top 5 songs and display them in ratio of how much the user listens to them?

The resulting page can be used as new Home page for the app

**Comment by z-huang** [comment]
Yes, I'll do something like this to replace the home tab.

**Reply by NeeRaj-2401** [reply]
Replace in the sense ? gonna totally remove the current home tab (ytm webview) or u r just planning to keep the current home tab under another section and implementing recommendation for Home Tab

**Reply by z-huang** [reply]
Yes, replace totally. The current home tab is not helpful to me. The recommendations are almost the same and I never listen to YouTube generated playlists. The new home tab in my mind includes most played songs, recommended songs and new released albums. No recommended albums, playlists and artists because they are less inducing for the user to have a click. I'll probably remove the home tab in 0.5.0 and let it return with new design in the future release.

**Reply by NeeRaj-2401** [reply]
But once we login the quick picks part was very useful for me
If something very similar to that quick picks, new albums and atleast artists ..along with the other things which u mentioned are introduced then it will be fine and useful to users.

Also will you remove the login to yt part too?

**Reply by z-huang** [reply]
Login feature won't be removed. It's only used for data migration and synchronization. Quick picks is not an intended feature because it won't update according to your listening history in InnerTune.
I won't browse an artist that I don't know. Only if I've heard his/her song will I be interested. Therefore, I'm not considering recommended artists in the home tab. Most importantly, I think the home tab is a place for the user to start a "song" conveniently, without too much tap.

**Comment by sumitmishra90** [comment]
Sir please add a sleep timer if possible 👍

**Reply by z-huang** [reply]
See #97.
